packages/storage/amplify_storage_s3_dart/test/storage_s3_service/task/s3_upload_task_test.dart

```diff
@@ -1094,8 +1094,10 @@ void main() {
           'should throw exception if the file to be upload is too large to initiate a multipart upload',
```

Comment:
question: Are the existing tests in this file sufficient to ensure that certain use cases result in expected number of parts? e.g. "file size x as a Stream results in n parts, each of size z." It looks like there are already checks for that ([example](https://github.com/aws-amplify/amplify-flutter/blob/next/packages/storage/amplify_storage_s3_dart/test/storage_s3_service/task/s3_upload_task_test.dart#L868)). Just checking if it's possible or makes sense to add checks for extra logic in this PR in `_startNextUploadPart`, in addition to the tests already here for the part size utils.

Reply:
Good call, I've added two new tests, which verifies the new branching added to `_startNextUploadPart ` by asserting calling `AWSFile.getChunkedStreamReader` vs `AWSFile.openRead`.